File: Codevita/MockVita1/A.cpp
#include <bits/stdc++.h>
#include <bits/stdc++.h>
using namespace std;        
#define forf(i,n) for(int i=0;i<(int)(n);i++)
#define fore(i,a,b) for(int i=(int)(a);(a>=b)?i>=b:i<=b;(a>=b)?i-=1:i+=1)
#define forb(i,n) for(int i=(int)(n-1);i>=0;i--)
typedef long long int int64;
typedef vector<int> vint;
typedef vector<int64> vint64;

int main()
{
	freopen("in.txt","r", stdin);
	char arr[35], c,q;
	cin >> arr >> q;
	int l= strlen(arr), start=0.n;
	char first[q];
	int i=0
	while (i<q){
		cin >> n>>c;
		if (c=='L') {
			start += (n%l);
		} else {
			start -= (n%l);
		}

			first[i] = arr[start];
			i++;
	}


}

File: Codevita/MockVita1/A.py
arr = input().strip()
q = int(input())

l = len(arr)
first = []
t=[]
start,i = 0,0
if q > l :
	print("NO")
else:
	while i < q:
		t = input().strip().split(" ")
		if t[0]=="L":
			start += (int(t[1])%l)
		else:
			start -= (int(t[1])%l)
		first.append(arr[start])
		i+=1

	#arr = sorted(arr)
	first = str(sorted(first))
	#print(arr, first)

	for i in range(l):
		if arr[i] in first:
			#print(arr[i:i+3])
			if str(sorted(arr[i:i+q])) == first:
				print("YES")
				exit(0)

	print("NO")